File: prime_numbers/benchmark.cs
using System;
using System.Collections.Generic;

public class Program
{
    private static List<int> get_primes7(int n)
    {
        List<int> res = new List<int>();
        if (n < 2) return res;

        if (n == 2)
        {
            res.Add(2);
            return res;
        }

        List<int> s = new List<int>();
        for (int z = 3; z < n + 1; z += 2)
        {
            s.Add(z);
        }

        int mroot = (int)Math.Sqrt(n);
        int half = s.Count;
        int i = 0;
        int m = 3;

        while (m <= mroot)
        {
            if (s[i] != 0)
            {
                int j = (int)((m * m - 3) / 2);
                s[j] = 0;
                while (j < half)
                {
                    s[j] = 0;
                    j += m;
                }
            }

            i = i + 1;
            m = 2 * i + 3;
        }

        res.Add(2);
        s.RemoveAll(item => item == 0);
        res.AddRange(s);

        return res;
    }

    public static void Main(string[] args)
    {
        Stopwatch stopWatch = new Stopwatch();
        stopWatch.Start();
        for (var i = 0; i < 10; i++) {
            var res = get_primes7(10000000);
            Console.Write("Found {0} prime numbers.\n", res.Count);
        }
        stopWatch.Stop();

        var totalSeconds = Math.Round(stopWatch.Elapsed.TotalSeconds, 3);
        Console.Write($"Execution time: {totalSeconds}\n");
    }
}

Program.Main(null);

File: prime_numbers/benchmark.rs
use std::time::Instant;

fn get_primes_7(n: usize) -> Vec<usize> {
    if n < 2 {
        return Vec::new();
    }
    if n == 2 {
        return vec![2];
    }

    let mut s: Vec<usize> = (3..=n).step_by(2).collect();

    let mut j: usize;
    let mut m = 3;
    let mroot = (n as f64).sqrt() as usize;
    let half = s.len();
    let mut i = 0;
    while m <= mroot {
        if s[i] != 0 {
            j = (m * m - 3) / 2;
            s[j] = 0;
            while j < half {
                s[j] = 0;
                j += m;
            }
        }
        i += 1;
        m = 2 * i + 3;
    }

    let mut res: Vec<usize> = Vec::new();
    res.push(2);
    for v in s.iter() {
        if *v != 0 {
            res.push(*v);
        }
    }

    res
}

fn main() {
    let start_time = Instant::now();

    for _ in 0..10 {
        let res = get_primes_7(10_000_000);
        println!("Found {} prime numbers.", res.len());
    }

    let duration = start_time.elapsed();
    println!("Execution time: {:?}", duration);
}
